File: berkelium-java/src/main/java/org/berkelium/demo/runtime/Runtime2.java
// Copyright (c) 2013 The Berkelium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

package org.berkelium.demo.runtime;

import org.berkelium.api.BerkeliumFactory;
import org.berkelium.api.Runtime;

public class Runtime2 {
	public static void main(String[] args) {
		// Create Runtime
		Runtime runtime = BerkeliumFactory.getInstance().createRuntime();
		/*GREEN*/// close runtime/*GREEN*/
		/*GREEN*/runtime.dispose();/*GREEN*/
	}
}

File: berkelium-cpp/src/demo/runtime/runtime2.cpp
// Copyright (c) 2013 The Berkelium Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include <Berkelium/API/Runtime.hpp>
#include <Berkelium/API/BerkeliumFactory.hpp>

int main(int argc, char* argv[])
{
	/*GREEN*/// no Runtime Instance yet/*GREEN*/
	{
		Berkelium::RuntimeRef runtime(Berkelium::BerkeliumFactory::createRuntime());
		/*GREEN*/// you can use the 'runtime' Variable here/*GREEN*/
	}
	/*GREEN*/// The 'runtime' Variable got out of scope - Runtime Instance was automaticly freed/*GREEN*/
}
